backend/api/quivr_api/modules/knowledge/service/knowledge_service.py

```diff
@@ -1,53 +1,126 @@
+from typing import List
 from uuid import UUID
 
 from quivr_api.logger import get_logger
-from quivr_api.modules.knowledge.dto.inputs import (
-    CreateKnowledgeProperties,
-    KnowledgeStatus,
-)
-from quivr_api.modules.knowledge.repository.knowledge_interface import (
-    KnowledgeInterface,
-)
-from quivr_api.modules.knowledge.repository.knowledges import KnowledgeRepository
+from quivr_api.modules.dependencies import BaseService
+from quivr_api.modules.knowledge.dto.inputs import (CreateKnowledgeProperties,
+                                                    KnowledgeStatus)
+from quivr_api.modules.knowledge.dto.outputs import DeleteKnowledgeResponse
+from quivr_api.modules.knowledge.entity.knowledge import KnowledgeDB
+from quivr_api.modules.knowledge.repository.knowledges import \
+    KnowledgeRepository
 from quivr_core.models import QuivrKnowledge as Knowledge
 
 logger = get_logger(__name__)
 
 
-class KnowledgeService:
-    repository: KnowledgeInterface
+class KnowledgeService(BaseService[KnowledgeRepository]):
+    repository_cls = KnowledgeRepository
 
-    def __init__(self):
-        self.repository = KnowledgeRepository()
+    async def add_knowledge(
+        self, knowledge_to_add: CreateKnowledgeProperties
+    ) -> Knowledge:
+        knowledge_data = knowledge_to_add.dict()
+        knowledge = KnowledgeDB(**knowledge_data)
 
-    def add_knowledge(self, knowledge_to_add: CreateKnowledgeProperties):
-        knowledge = self.repository.insert_knowledge(knowledge_to_add)
-
-        return knowledge
+        inserted_knowledge_db_instance = await self.repository.insert_knowledge(
+            knowledge
+        )
 
-    def get_all_knowledge(self, brain_id: UUID):
-        knowledges = self.repository.get_all_knowledge_in_brain(brain_id)
+        assert inserted_knowledge_db_instance.id, "Knowledge ID not generated"
+        if inserted_knowledge_db_instance.source == "local":
+            source_link = f"s3://quivr/{inserted_knowledge_db_instance.brain_id}/{inserted_knowledge_db_instance.id}"
```

Comment:
The knowledge service shouldn't have the path logic. or it could have this logic if it has a storage instance.